grpc-server/src/main/scala/com/wavesplatform/api/grpc/TransactionsApiGrpcImpl.scala

```diff
@@ -102,7 +101,12 @@ class TransactionsApiGrpcImpl(blockchain: Blockchain, commonApi: CommonTransacti
           .map(_ => TransactionStatus(txId, TransactionStatus.Status.UNCONFIRMED))
           .orElse {
             commonApi.transactionById(txId.toByteStr).map { m =>
-              val status = if (m.succeeded) ApplicationStatus.SUCCEEDED else ApplicationStatus.SCRIPT_EXECUTION_FAILED
+              val status = m.status match {
+                case TxMeta.Status.Succeeded => ApplicationStatus.SUCCEEDED
+                case TxMeta.Status.Failed    => ApplicationStatus.SCRIPT_EXECUTION_FAILED
+                case TxMeta.Status.Elided    => ApplicationStatus.ELIDED
+              }
```

Comment:
This appears 5 times. Let's create a function for this.

Reply:
Fixed